packages/roosterjs-content-model-core/lib/editor/Editor.ts

```diff
@@ -200,13 +200,14 @@ export class Editor implements IEditor {
      * Add a single undo snapshot to undo stack
      * @param entityState @optional State for entity if we want to add entity state for this snapshot
      */
-    takeSnapshot(entityState?: EntityState): Snapshot | null {
+    takeSnapshot(entityState?: EntityState, additionalState?: string[]): Snapshot | null {
         const core = this.getCore();
 
         return core.api.addUndoSnapshot(
             core,
             false /*canUndoByBackspace*/,
-            entityState ? [entityState] : undefined
+            entityState ? [entityState] : undefined,
+            additionalState
```

Comment:
Do we need both allowing additional state when calling the Editor API and also let Plugins handle the event to add more state?

I feel only adding snapshot state from the editor API would make the logic easier. If multiple plugins add different states could make handling of the state a bit harder to predict.